### ISSUE TITLE
Rule proposal: `perfer-async-generator-function`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2481]
### Description

Instead of return an object with `Symbol.asyncIterator`, async generator function syntax should be used directly.

### Fail

```js
function foo() {
	return {
		[Symbol.asyncIterator]() {
			yield 1;
		}
	}
}
```

### Pass

```js
async function * foo() {
  yield 1;
}
```

### Proposed rule name

perfer-async-generator-function

### Additional Info

I'm trying to use [`request-animation-frames`](https://github.com/sindresorhus/request-animation-frames/pull/2), but I found there are other modules using the same pattern,

**Comment by fisker** [comment]
Reason https://github.com/sindresorhus/request-animation-frames/pull/2#issuecomment-2419129537
